src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,707 @@
+    // is necessary for the generator but it does not affect the advanceable output
+    generator.setTerminalHalfSwitchTime(1.0);
+
+    // Due to how the generator works, the start time must be bigger than last impact time
+    const double startLeft = m_pImpl->left->getSteps().front().impactTime;
+    const double startRight = m_pImpl->right->getSteps().front().impactTime;
+    const double startTime = std::max(startLeft, startRight);
+
+    // Compute the contact states using the generator
+    if (!generator.generateFromFootPrints(m_pImpl->left,
+                                          m_pImpl->right,
+                                          startTime,
+                                          m_pImpl->dt.planner))
+    {
+        cleanup();
+        log()->error("{} Failed to generate from footprints.", logPrefix);
+        return false;
+    }
+
+    // Get the contact states over the horizon
+    std::vector<bool> leftStandingPeriod;
+    std::vector<bool> rightStandingPeriod;
+    generator.getFeetStandingPeriods(leftStandingPeriod, rightStandingPeriod);
+
+    // Bind dt to catch it in the next lambda
+    const auto& dt = m_pImpl->dt.planner;
+
+    // Lambda to convert Step to Contact, filling only the timings.
+    // Transforms will be included in a later stage.
+    auto convertStepsToContacts
+        = [dt](const std::vector<bool>& isFootInContactVector,
+               const StepList& steps) -> std::vector<Contacts::PlannedContact> {
+        std::vector<Contacts::PlannedContact> contacts;
+
+        for (const auto& step : steps)
+        {
+            auto contact = Contacts::PlannedContact();
+            contact.name = step.footName;
+            contact.activationTime = step.impactTime;
+            contact.deactivationTime = contact.activationTime;
+            contacts.push_back(contact);
+        }
+
+        size_t contactIdx = 0;
+        double activeTime = 0.0;
+
+        for (size_t idx = 1; idx < isFootInContactVector.size(); ++idx)
+        {
+            // Get the active contact
+            auto& contact = contacts[contactIdx];
+
+            const bool thisState = isFootInContactVector[idx];
+            const bool lastState = isFootInContactVector[idx - 1];
+
+            // Increase the active time
+            activeTime += dt;
+
+            // During impact, reset the active time counter
+            if (lastState == 0 && thisState == 1)
+                activeTime = 0.0;
+
+            // During lift, store the time in the active contact and get the new contact
+            if (lastState == 1 && thisState == 0)
+            {
+                contact.deactivationTime = contact.activationTime + activeTime;
+                contactIdx++;
+            }
+        }
+
+        // The deactivation time of the last contact has not yet been processed
+        contacts.back().deactivationTime = std::numeric_limits<float>().max();
```

Comment:
In this way you are setting the end time of only one contact to ``inf``, but what about the other foot?

Reply:
This line is in a lambda that is applied to both feet right below its definition

Reply:
Ah I see, cool!

Reply:
Operations for the two feet are the same, and introducing new private methods is a but cumbersome. I typically prefer keep code as local as possible, hence the lambdas (that can also capture some local variable, simplifying the logic).

Reply:
This is the only thing I'm scared of